### ISSUE TITLE
New login page (since 1.2) has broken DB 'location' selection

### ISSUE THREAD

**Issue by PDS-GH** [lirantal/daloradius#248]
Since release 1.2, the location from login page which allows to choose the DB location from array is broken. It always selects the default DB configured in daloradius.conf.php

In 1.1-X it properly selects the DB backend and shows the associated name as the location in the top right corner once logged in.
In 1.2 or 1.3 it always selects the default DB and shows 'default' as the location in the top right corner once logged in.

Extract of daloradius.conf.php

```
$configValues['CONFIG_DB_ENGINE'] = 'mysqli';
$configValues['CONFIG_DB_HOST'] = '192.168.0.1';
$configValues['CONFIG_DB_PORT'] = '3306';
$configValues['CONFIG_DB_USER'] = 'radius';
$configValues['CONFIG_DB_PASS'] = 'xxxx';
$configValues['CONFIG_DB_NAME'] = 'radius';

$configValues['CONFIG_LOCATIONS'] = array (
  "radius-db1" =>
  array (
    "Engine" => "mysqli",
    "Username" => "radius",
    "Password" => "xxxx",
    "Database" => "radius",
    "Hostname" => "192.168.0.1"
  ),
  "radius-db2" =>
  array (
    "Engine" => "mysqli",
    "Username" => "radius",
    "Password" => "xxxx",
    "Database" => "radius",
    "Hostname" => "192.168.0.2"
  ),
  "radius-db3" =>
  array (
    "Engine" => "mysqli",
    "Username" => "radius",
    "Password" => "xxxx",
    "Database" => "radius",
    "Hostname" => "192.168.0.3"
  )
);
```

**Comment by PDS-GH** [comment]
@MiguelVis could it be related to #210 / #211 ?

**Comment by lirantal** [comment]
Good find @PDS-GH, looks like that's related but I haven't looked at the code changes yet to say why.

**Comment by lirantal** [comment]
I think the issue is related to prior change to `login.php` like this one: https://github.com/MiguelVis/daloradius/commit/2db45ecc6f9817c5a9721d0f80c4348849d88c0f#diff-b5866acb4c2b0d900c2623c453f9cf056193de7d29b280e8a9ceb7ef94bd28ad

That was a very big PR that we landed and could've been related to the breaking change.

**Comment by MiguelVis** [comment]
@lirantal 
@PDS-GH 

It's documented in fix #251 but just for the record I will repeat here:

> The previous code failed to read the db configuration for a given location.
> 
> Also it failed to select the db port in case it was not set.
> 
> Now it's compulsory to set the db port for each location in the configuration file, in order to be consistent with the default configuration.
> 
> The configuration sample file has been modified to reflect this change:

```
// Locations Configuration directives
// Locations directives are support for accessing different databases from the daloRADIUS Login console
// adjust the locations below for databases you are running (if you are running more than one).
// You could configure invoice templates for each location - optional.
$configValues['CONFIG_LOCATIONS'] = array(

        "Location Example 1" => array(
                "Engine"   => "mysql",
                "Username" => "root",
                "Password" => "",
                "Database" => "radius",
                "Hostname" => "127.0.0.1",
                "Port"     => "3306",
                "CONFIG_INVOICE_TEMPLATE" => "invoice_template_location_1.html",
                "CONFIG_INVOICE_ITEM_TEMPLATE" => "invoice_item_template_location_1.html"
        ),
        "Location Example 2" => array(
                "Engine"   => "mysql",
                "Username" => "db_usertest",
                "Password" => "db_passtest",
                "Database" => "test_db1",
                "Hostname" => "localhost",
                "Port"     => "3306"
        )
);
```

@PDS-GH please add the "Port" key to every location in your daloradius.conf.php file as explained.

**Comment by PDS-GH** [comment]
It now works as expected, awesome!
Thanks a lot @MiguelVis !
